src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -0,0 +1,262 @@
+package org.folio.auth.authtokenmodule.tokens;
+
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.JsonObject;
+import java.util.Base64;
+import com.nimbusds.jose.JOSEException;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.auth.authtokenmodule.BadSignatureException;
+import org.folio.auth.authtokenmodule.TokenCreator;
+import org.folio.okapi.common.XOkapiHeaders;
+import java.text.ParseException;
+
+/**
+ * An abstract class that all token types should extend, with one required method
+ * that all must implement to handle their validation logic. This class also exposes
+ * a handful of static methods which provide convenient ways of working with tokens,
+ * including a method to parse and validate a token in one step.
+ */
+public abstract class Token {
+  private static final String CALLING_MODULE_HEADER = "X-Okapi-Calling-Module";
```

Comment:
This header has been there for a long time. Okapi doesn't set it. Never has. I think we can remove it (and handling for it).

Reply:
Yes, not used anywhere: https://github.com/search?q=org%3Afolio-org+X-Okapi-Calling-Module&type=code

Reply:
Will remove.